closed/Components.gmk

```diff
@@ -0,0 +1,71 @@
+# (c) Copyright IBM Corp. 2020, 2022 All Rights Reserved
+# ===========================================================================
+# This code is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 only, as
+# published by the Free Software Foundation.
+#
+# IBM designates this particular file as subject to the "Classpath" exception
+# as provided by IBM in the LICENSE file that accompanied this code.
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, see <http://www.gnu.org/licenses/>.
+# ===========================================================================
+
+no_default :
+	$(error Components.gmk has no default target)
+
+include $(SPEC)
+include $(TOPDIR)/make/common/MakeBase.gmk
+
+.PHONY : no_default stage-components
+
+# copy_component_tree
+# -------------------
+# Copy recursively source $2 to target directory $1.
+# $1 = target directory
+# $2 = source directory
+define copy_component_tree
+	$(if $(filter 1,$(words $2)),,$(error copy_component_tree requires exactly one source directory; got '$2'))
+	@$(MKDIR) -p $1
+	@$(CP) -fR $2 $1
+endef
+
+# download_icc_binaries
+# -------------------
+# $1 = icc binaries url
+# $2 = icc sdk binaries url
+# $3 = icc binaries credential
+define download_icc_binaries
+	@$(MKDIR) -p $(OPENJCEPLUS_TOPDIR)/icc/jgsk_sdk/lib64/
+	$(if $(strip $3),
+        @curl -H "Authorization: Bearer $3" -H "Accept: application/octet-stream" -L $1 --output $(OPENJCEPLUS_TOPDIR)/icc/jgsk_crypto.tar, \
+        @echo "GSKIT_BIN_CREDENTIAL is either undefined or empty, will add curl command HERE without credential.")
+	$(if $(strip $3),
+        @curl -H "Authorization: Bearer $3" -H "Accept: application/octet-stream" -L $2 --output $(OPENJCEPLUS_TOPDIR)/icc/jgsk_crypto_sdk.tar, \
+        @echo "GSKIT_BIN_CREDENTIAL is either undefined or empty, will add curl command HERE without credential.")
+	@$(TAR) -xf $(OPENJCEPLUS_TOPDIR)/icc/jgsk_crypto_sdk.tar -C $(OPENJCEPLUS_TOPDIR)/icc/
+	@$(TAR) -xf $(OPENJCEPLUS_TOPDIR)/icc/jgsk_crypto.tar -C $(OPENJCEPLUS_TOPDIR)/icc/jgsk_sdk/lib64/
+endef
+
+ifeq (true,$(BUILD_JGSKIT))
+
+.PHONY : stage-openjceplus
+
+stage-components : stage-openjceplus
+
+# Copy openjceplus java source codes into $(TOPDIR)/src/openjceplus folder
+# Download ICC binaries into $(OPENJCEPLUS_TOPDIR)/icc folder
+stage-openjceplus :
+	@$(ECHO) Staging OpenJCEPlus in $(TOPDIR)/src/openjceplus
+	$(call copy_component_tree,$(TOPDIR)/src/openjceplus/share/classes,$(OPENJCEPLUS_TOPDIR)/src/main/java/*)
```

Comment:
This is not reasonable: it will pollute the working directory.

Reply:
@keithc-ca I appended the OPENJCEPLUS_TOPDIR folder to TOP_SRC_DIRS in Modules.gmk, as below. I can see its java source codes and native codes built and in its "target" folder. But we would like the OpenJCEPlus java code built as a jmod file in JDK's "jmod" folder. Do you have any idea how we can do that?

```
TOP_SRC_DIRS += \
	$(J9JCL_SOURCES_DIR) \
	$(OPENJCEPLUS_TOPDIR) \
	#
```

Reply:
The source in `$(OPENJCEPLUS_TOPDIR)` must be organized as expected by openjdk, with the same structure you'll find under `$(TOPDIR)/src`:
```
openjceplus/share/classes/module-info.java
openjceplus/share/classes/...         (other java code, etc.)
openjceplus/share/native/...          (if needed)
```

Reply:
Copied the java codes into "$TOPDIR/OpenJCEPlus/src/main/openjceplus/share/classes/", and appended "$TOPDIR/OpenJCEPlus/src/main/" folder to "TOP_SRC_DIRS". The java codes built success and jmod file generated.